python/tvm/relay/ir_pass.py

```diff
@@ -491,9 +493,18 @@ def gradient(expr, mod=None):
     mod : Optional[tvm.relay.Module]
         The global module.
 
+    mode : Optional[String]
+        The mode of the automatic differentiation algorithm.
+        'first_order' only work on first order code, but will not produce reference nor closure.
+        'higher_order' work on all code using reference and closure.
```

Comment:
A very basic question: what exactly 'higher order' means here?
Aren't we supposed to supply "second order" or "third order" as an argument?

Reply:
higher order mean it work on higher order program (program with closure)

Reply:
Sounds cool:)

Reply:
@masahi technical its higher order, higher order AD or as we have been jokingly calling it Santa Clause AD.

You can implement a helper on top which applies higher order an arbitrary number of times to get nth order gradients.